src/Processors/Transforms/MemoryBoundMerging.h

```diff
@@ -0,0 +1,216 @@
+#pragma once
+
+#include <Core/SortDescription.h>
+#include <Interpreters/sortBlock.h>
+#include <Processors/IProcessor.h>
+#include <Processors/Transforms/AggregatingTransform.h>
+
+#include <Poco/Logger.h>
+#include <Common/logger_useful.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+}
+
+
+/// Has several inputs and single output.
+/// Read from inputs merged buckets with aggregated data, sort them by bucket number and block number.
+/// Presumption: inputs return chunks with increasing bucket and block number, there is at most one chunk with the given bucket and block number.
+class SortingAggregatedForMemoryBoundMergingTransform : public IProcessor
```

Comment:
this transform will be used for new merging algorithm implementation for ordinary aggregation too, but so far I'm not sure that it works correctly in this case.